api/v1alpha4/types.go

```diff
@@ -106,11 +106,27 @@ type VnetSpec struct {
 	// +optional
 	CIDRBlocks []string `json:"cidrBlocks,omitempty"`
 
+	// Peerings defines a list of peerings of the newly created virtual network with existing virtual networks.
+	// +optional
+	Peerings VnetPeerings `json:"peerings,omitempty"`
```

Comment:
TODO: add conversion and webhook validation (using https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-manage-peering#requirements-and-constraints)